message_set.go

```diff
@@ -48,9 +48,11 @@ func (msb *MessageBlock) decode(pd packetDecoder) (err error) {
 type MessageSet struct {
 	PartialTrailingMessage bool // whether the set on the wire contained an incomplete trailing MessageBlock
 	Messages               []*MessageBlock
+	UnderlyingMessageCount int // in case of compressed messages wraps in a single fake "message"
```

Comment:
You should still be able to ask for `len(set.Messages[0].Msg.Set.Messages)`? I'm not sure why you even need this value to be honest?